### PR TITLE
sa-4 : 4 : all-changes-till-now

### PR DESCRIPTION
**Pull request by Smeetiiitdmj**
## Issue that this pull request solves

Closes: #1013

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Other (please describe):

## Checklist

-   [x] My code follows the [style guidelines of this project](https://docs.google.com/document/d/1GI2Hile8UbGZ82gFe1y4wAVPcNPXip1cmXTzog1S41U/edit)
-   [x] I have performed a self-review of my own code
-   [x] I have created new branch for this pull request
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] My changes does not break the current system and it passes all the current test cases.

### CODE REVIEW

**Review by rishiktejatanugula — COMMENTED** [review]
Approved @akshatnema

File: FusionIIIT/applications/placement_cell/views.py [hunk 1]
```diff
@@ -2608,17 +2609,30 @@ def cv(request, username):
...
+    # if int(str(profile.id)[:2]) == 20:
+    #     if (now.month>4):
+    #       roll = 1+now.year-int(str(profile.id)[:4])
+    #     else:
+    #       roll = now.year-int(str(profile.id)[:4])
+    # else:
+    #     if (now.month>4):
+    #       roll = 1+(now.year)-int("20"+str(profile.id)[0:2])
+    #     else:
+    #       roll = (now.year)-int("20"+str(profile.id)[0:2])
```
Comment (akshatnema) [inline FusionIIIT/applications/placement_cell/views.py]:
```suggestion
    # if int(str(profile.id)[:2]) == 20:
    #     if (now.month>4):
    #       roll = 1+now.year-int(str(profile.id)[:4])
    #     else:
    #       roll = now.year-int(str(profile.id)[:4])
    # else:
    #     if (now.month>4):
    #       roll = 1+(now.year)-int("20"+str(profile.id)[0:2])
    #     else:
    #       roll = (now.year)-int("20"+str(profile.id)[0:2])

```

Kindly remove these comments if not needed.

File: FusionIIIT/applications/placement_cell/views.py [hunk 2]
```diff
@@ -2608,17 +2609,30 @@ def cv(request, username):
     # print(achievementcheck,' ',educationcheck,' ',publicationcheck,' ',patentcheck,' ',internshipcheck,' ',projectcheck,' \n\n\n')
     user = get_object_or_404(User, Q(username=username))
     profile = get_object_or_404(ExtraInfo, Q(user=user))
+    student_info=get_object_or_404(Student,Q(id=user.username))
+    # print("Student------>",vars(student_info))
+    batch=student_info.batch
+    # print("Year----->",user.date_joined.year)
+    # print("User----->",vars(user))
+    # print("Profile---->",vars(profile))
```
Comment (akshatnema) [inline FusionIIIT/applications/placement_cell/views.py]: same goes for the comments in these lines too.

File: requirements.txt [hunk 3]
```diff
@@ -61,14 +61,14 @@ requests==2.25.1
...
+urllib3==1.26.2
 vine==5.0.0
 wcwidth==0.2.5
 webencodings==0.5.1
 whitenoise==5.2.0
 xhtml2pdf==0.2.5
 xlrd==2.0.1
 XlsxWriter==1.3.7
-xlwt==1.3.0
+xlwt==1.3.0
```
Comment (akshatnema) [inline requirements.txt]: The changes in the requirements file are not acceptable. Kindly remove these changes from the file.

File: FusionIIIT/templates/placementModule/placementstatistics.html [hunk 4]
```diff
@@ -357,32 +357,32 @@ <h4 class="ui image header">
...
-                                            <td style="border:1px solid black;">{{ rec.test_type }}</td>
-                                            <td>
+                                            <td style="border:1px solid black; text-align:center">{{ rec.id }}</td>
+                                            {% comment %} <td style="border:1px solid black;">{{ rec.placement_type }}</td> {% endcomment %}
+                                            <td style="border:1px solid black; text-align:center">{{ rec.name }}</td>
+                                            <td style="border:1px solid black; text-align:center">{{ rec.ctc }}</td>
+                                            {% comment %} <td style="border:1px solid black; text-align:center">{{ rec.year }}</td> {% endcomment %}
+                                            {% comment %} <td style="border:1px solid black;">{{ rec.test_score }}</td> {% endcomment %}
+                                            {% comment %} <td style="border:1px solid black;">{{ rec.test_type }}</td> {% endcomment %}
+                                            {% comment %} <td>
```
Comment (akshatnema) [inline FusionIIIT/templates/placementModule/placementstatistics.html]: Why so many lines in the UI are commented in the HTML files?

File: FusionIIIT/templates/placementModule/placementstatistics.html [hunk 5]
```diff
@@ -357,32 +357,32 @@ <h4 class="ui image header">
...
+                                            <th style="border:1px solid black; ">Roll No.</th>
+                                            {% comment %} <th style="border:1px solid black; ">Placement Type</th> {% endcomment %}
                                             <th style="border:1px solid black;">Name</th>
                                             <th style="border:1px solid black; ">CTC</th>
-                                            <th style="border:1px solid black; ">Year</th>
-                                            <th style="border:1px solid black;">Test Score</th>
-                                            <th style="border:1px solid black;">Test Type</th>
+                                            {% comment %} <th style="border:1px solid black; ">Year</th> {% endcomment %}
+                                            {% comment %} <th style="border:1px solid black;">Test Score</th>
+                                            <th style="border:1px solid black;">Test Type</th> {% endcomment %}
```
Comment (akshatnema) [inline FusionIIIT/templates/placementModule/placementstatistics.html]: Same goes for here too.

File: FusionIIIT/applications/placement_cell/models.py [hunk 6]
```diff
@@ -330,6 +330,8 @@ def __str__(self):
 
 
 class PlacementRecord(models.Model):
+    # id = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE, primary_key=True)
+    # id = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE, primary_key=True)
```
Comment (akshatnema) [inline FusionIIIT/applications/placement_cell/models.py]:
```suggestion
    # id = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE, primary_key=True)
    # id = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE, primary_key=True)

```

remove the code comments if not needed.

File: FusionIIIT/applications/placement_cell/models.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # imports
 import datetime
-
+from applications.globals.models import ExtraInfo
```
Comment (akshatnema) [inline FusionIIIT/applications/placement_cell/models.py]: the module has been imported but not used.

**Review by akshatnema — CHANGES_REQUESTED** [review]
Kindly answer the queries made.

File: requirements.txt (hunk 3 above)
Comment (akshatnema) [reply requirements.txt]: @Smeetiiitdmj These changes are not removed yet. Kindly remove these changes.

**Review by Smeetiiitdmj — COMMENTED** [review]
Changes are committed

File: requirements.txt (hunk 3 above)
Comment (Smeetiiitdmj) [reply requirements.txt]: Kindly check now, I have committed

**Comment by akshatnema** [comment]
@SmeetChavan this is a duplicate PR. changes are already merged in sa-4 via #1115. Closing this PR.

**Pull request closed without merging**